src/main/java/reactor/kafka/receiver/internals/ConsumerEventLoop.java

```diff
@@ -218,30 +223,46 @@ public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
 
         private final Duration pollTimeout = receiverOptions.pollTimeout();
 
+        private final AtomicBoolean pausedByUs = new AtomicBoolean();
+
+        private final Set<TopicPartition> pausedByUser = new HashSet<>();
+
+        private final AtomicBoolean scheduled = new AtomicBoolean();
+
         @Override
         public void run() {
             try {
+                this.scheduled.set(false);
                 if (isActive.get()) {
                     // Ensure that commits are not queued behind polls since number of poll events is
                     // chosen by reactor.
                     commitEvent.runIfRequired(false);
                     long r = requested;
                     if (r > 0) {
                         if (!awaitingTransaction.get()) {
-                            consumer.resume(consumer.assignment());
+                            if (pausedByUs.getAndSet(false)) {
+                                Set<TopicPartition> toResume = new HashSet<>(consumer.assignment());
+                                toResume.removeAll(this.pausedByUser);
+                                this.pausedByUser.clear();
+                                consumer.resume(toResume);
+                                log.debug("Resumed");
+                            }
                         } else {
-                            consumer.pause(consumer.assignment());
-                            schedule();
+                            if (!pausedByUs.getAndSet(true)) {
+                                this.pausedByUser.addAll(consumer.paused());
+                                consumer.pause(consumer.assignment());
+                                log.debug("Paused 1");
+                            }
                         }
-                    } else {
+                    } else if (!pausedByUs.getAndSet(true)) {
+                        this.pausedByUser.addAll(consumer.paused());
                         consumer.pause(consumer.assignment());
+                        log.debug("Paused 2");
```

Comment:
What the info does it bring for end-users in the logs?
I mean this one and the previous `Paused 1` . 
Doesn't look like very useful...